File: Miscellaneous/palsquare.py
"""
ID: alexzha5
LANG: PYTHON3
TASK: palsquare
"""
with open('palsquare.in', 'r') as fin:
    base = int(fin.readline().strip())

def base_change(num, base):
    """Change ``num'' to given base
    Upto base 36 is supported."""

    converted_string, modstring = "", ""
    currentnum = num
    if not num:
        return '0'
    while currentnum:
        mod = currentnum % base
        currentnum = currentnum // base
        if chr(48 + mod + 7 * (mod > 10)) == ":":
            converted_string = "A" + converted_string
        else:
            converted_string = chr(48 + mod + 7 * (mod > 10)) + converted_string
    return converted_string

def is_pal(a):
    return a[::-1] == a

fout = open('palsquare.out', 'w')
for i in range(1, 301):
    b = base_change(i * i, base)
    if is_pal(b):
        fout.write(base_change(i, base) + " " + b + "\n")
fout.close()

File: Miscellaneous/palsquare.cpp
/*
ID: alexzha5
TASK: palsquare
LANG: C++
*/

#include <bits/stdc++.h>

using namespace std;

map<int, int> squares;
int base;

int base_change(int num, int base){
  if(base == 10)
    return num;
  int digit = 1, ret = 0, b = base;
  for(; num != 0; digit *= 10){
    ret += (num % base) * digit;
    num /= base;
  }
  return ret;
}

bool is_pal(int b){
  //reverse b
  int rev, copy = b;
  for(; b != 0; b /= 10){
    rev += rev * 10 + b % 10;
  }
  return copy == rev;
}

int main() {
    ofstream fout ("palsquare.out");
    ifstream fin ("palsquare.in");
    fin >> base;
    int b;
    for(int i = 1; i <= 300; i++)
      squares[i] = i * i;

    for(int i = 1; i <= 300; i++){
      b = base_change(squares[i], base);
      if(is_pal(b)){
        fout << base_change(i, base) << " " << b << endl;
      }
    }
    return 0;
}
